From ben.hutchings@codethink.co.uk  Wed Jan 24 10:19:07 2018
From: Ben Hutchings <ben.hutchings@codethink.co.uk>
Date: Wed, 24 Jan 2018 02:31:19 +0000
Subject: x86/microcode/intel: Fix BDW late-loading revision check
To: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Cc: stable@vger.kernel.org
Message-ID: <20180124023119.kaendz4jiuejowxr@xylophone.i.decadent.org.uk>
Content-Disposition: inline

From: Ben Hutchings <ben.hutchings@codethink.co.uk>

The backport of commit b94b73733171 ("x86/microcode/intel: Extend BDW
late-loading with a revision check") to 4.4-stable deleted a "return true"
statement.  This bug is not present upstream or other stable branches.

Signed-off-by: Ben Hutchings <ben.hutchings@codethink.co.uk>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 arch/x86/kernel/cpu/microcode/intel.c |    1 +
 1 file changed, 1 insertion(+)

--- a/arch/x86/kernel/cpu/microcode/intel.c
+++ b/arch/x86/kernel/cpu/microcode/intel.c
@@ -1005,6 +1005,7 @@ static bool is_blacklisted(unsigned int
 	    c->microcode < 0x0b000021) {
 		pr_err_once("Erratum BDF90: late loading with revision < 0x0b000021 (0x%x) disabled.\n", c->microcode);
 		pr_err_once("Please consider either early loading through initrd/built-in or a potential BIOS update.\n");
+		return true;
 	}
 
 	return false;
